File: Results/DER/Translation/translation_output/no_test/granite/codenet/python/java/p02897.java
System:
You are an expert java programmer and assistant

Question:
Translate the following code from python to java and enclose your solution inside ```java```:
```
N = int(input())
print(-(-N//2)/N)
```


Answer:
```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        double result = -(-N/2) / (double) N;
        System.out.println(result);
    }
}
```<|endoftext|>

File: Results/DER/Translation/translation_output/no_test/granite/codenet/java/python/p02897.py
System:
You are an expert python programmer and assistant

Question:
Translate the following code from java to python and enclose your solution inside ```python```:
```
import java.util.Scanner;

public class p02897 {
  public static void main(String[] args) {

    Scanner sc = new Scanner(System.in);

    double n = sc.nextDouble();

    double odd = 0;
    double notOdd = 0;
    for (int i = 1; i <= n; i++) {
      if (i % 2 == 0) {
        notOdd++;
      } else {
        odd++;
      }
    }
    System.out.println(odd / n);
  }
}

```


Answer:
```python
import sys

n = float(input())

odd = 0
notOdd = 0
for i in range(1, n+1):
    if i % 2 == 0:
        notOdd += 1
    else:
        odd += 1

print(odd / n)
```<|endoftext|>
